File: types/tag.ts
import type { Blog } from "./blog";
import type { CommonModel, ListReq } from "./common";

export interface Tag extends CommonModel {
  name: string;
  slug: string;
  description: string;
  blogCount: number;
  blogs?: Blog[];
}

export interface TagListReq extends ListReq {
  name?: string;
  slug?: string;
  sortBy?: "createdAt" | "updatedAt";
  order?: "asc" | "desc";
}

export interface TagListResp {
  total: number;
  lists: Tag[] | null;
}

export interface TagCreateReq {
  name: string;
  slug: string;
  description?: string;
}

File: internal/model/tag.go
package model

type Tag struct {
	CommonModel

	Name        string `gorm:"size:255;unique;not null;comment:标签名称" json:"name"`
	Slug        string `gorm:"size:255;uniqueIndex;not null;comment:标签 slug" json:"slug"`
	Description string `gorm:"size:255;comment:标签描述" json:"description"`

	Blogs []*Blog `gorm:"many2many:blog_tags;" json:"blogs,omitempty"`
}
